### PR TITLE
Improvements around IAM policies

### PR DESCRIPTION
**Pull request by farski**
This pull request refactors a number of identical IAM policies out of individually-defined roles, and into shared managed policies. Key things to look for are to ensure that anywhere an inline policy was removed, a corresponding (and correct) managed policy was added to that role.

The changes should be one-to-one in terms of the actual permissions any given role has before and after, with the exception of one case where there were two polices doing the same thing but in different ways; I made them the same.

I also pulled out the `Path` property from all the policies, because `/` is the default value, so they were unnecessary.

Documentation was also added.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,6 +22,14 @@ The job input format should generally avoid using input formats specific to part
...
 
+### S3 Access Permissions
+
+There are two common reasons that a given state's resource (like a Lambda function or Fargate task) would need access to S3: reading from the artifact bucket, and writing new files to the destination buckets. There are two managed IAM policies available within the CloudFormation stack that provide those two levels off access: `ArtifactBucketReadOnlyAccessPolicy` and `DestinationBucketsWriteAccessPolicy`. You can include them in the list of `ManagedPolicyArns` for IAM roles you create for your stack resources.
+
+The `DestinationBucketsWriteAccessPolicy` provides write access to a set of resources that is defined when the Porter CloudFormation stack is launched. Those resources are defined in the stack's parameters. It takes two parameters because the policy includes both bucket-level permissions (e.g., `arn:aws:s3:::myBucket`) and object-level permissions (e.g., `arn:aws:s3:::myBucket/*`), and there's no good way to generate the wildcard versions from a list of non-wildcard values.
```
Comment (cavis) [inline CONTRIBUTING.md]: > no good way to generate the wildcard versions from a list of non-wildcard values.

Wish list: CFN regex-replace.

File: template.yml [hunk 2]
```diff
@@ -1195,20 +1129,24 @@ Resources:
...
-          PolicyDocument:
-            Version: "2012-10-17"
-            Statement:
-              - Effect: Allow
-                Action:
-                  - s3:Get*
-                  - s3:Put*
-                Resource:
-                  - !Sub ${ArtifactBucket.Arn}/*
+        # TODO Need to make sure the ArtifactBucketWriteAccessPolicy policy
```
Comment (cavis) [inline template.yml]: So AWS transcribe might need read and/or list access to the destination?

File: CONTRIBUTING.md (hunk 1 above)
Comment (farski) [reply CONTRIBUTING.md]: @cavis I actually did figure this out, if you do something like

`Split (',', Join('', [Join(',/*', bucket_list), '/*']))`

you end up with a list of wildcards, but you lose the ability to limit access to a specific prefix within the bucket, so I decided not to do it.

File: template.yml (hunk 2 above)
Comment (farski) [reply template.yml]: I can't remember if I gave it `put*` and `get*` because it needed it, or because I was being lazy

**Pull request merged**